src/chord/channels.cljc

```diff
@@ -50,3 +52,20 @@
       (p/close! write-ch)
       (when on-close
         (on-close)))))
+
+(defn- on-close [ws ws-ch]
+  #?(:clj  (http/on-close ws (fn [_] (close! ws-ch)))
+     :cljs (.on ws "close" #(close! ws-ch))))
+
+(defn wrap-websocket [socket {:keys [read-ch write-ch] :as opts} & [cleanup]]
```

Comment:
can you coalesce the opts map and the cleanup function, so that we don't have multiple options parameters? e.g. `(defn wrap-websocket [socket {:keys [read-ch write-ch on-close] :as opts}] ...)`

Reply:
After a bit more thought, I wonder why we leave this up to the implementer at all? We only have two types of connections, so I can clean up automatically.